### PR TITLE
[X86][AVX10] Allow AVX10 use VBMI2 instructions

### PR DESCRIPTION
**Pull request by phoebewang**

### CODE REVIEW

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 1]
```diff
@@ -1970,20 +1970,27 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
...
   }// useAVX512Regs
 
+  if (Subtarget.hasVBMI2()) {
+    for (auto VT : {MVT::v8i16, MVT::v4i32, MVT::v2i64, MVT::v16i16, MVT::v8i32,
+                    MVT::v4i64}) {
+      setOperationAction(ISD::FSHL, VT, Custom);
+      setOperationAction(ISD::FSHR, VT, Custom);
+    }
+
+    setOperationAction(ISD::ROTR, MVT::v8i16, Custom);
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86ISelLowering.cpp:1990]: How about ROTL for v16i16?

Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:1990]: Good question. Actually both were set to `Custom` already. So remove them.

**Review by KanRobert — APPROVED** [review]
LGTM

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 2]
```diff
@@ -1970,20 +1970,24 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
...
       }
 
       setOperationAction(ISD::ROTL, MVT::v32i16, Custom);
-      setOperationAction(ISD::ROTR, MVT::v8i16,  Custom);
-      setOperationAction(ISD::ROTR, MVT::v16i16, Custom);
       setOperationAction(ISD::ROTR, MVT::v32i16, Custom);
     }
   }// useAVX512Regs
 
+  if (Subtarget.hasVBMI2()) {
```
Comment (e-kud) [inline llvm/lib/Target/X86/X86ISelLowering.cpp:1983]: Now `setOperationAction` is called without `!Subtarget.useSoftFloat()`. I've just realized that we have this predicate on integer operations. I don't know the reasons why it is so. I assume it is used as a vector register availability predicate. If so, then we need it here as well

Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:1983]: Thanks for the reminder!
I think `useSoftFloat` only matters when calling to `addRegisterClass`. We don't have any chance to lower the opcode with a legal type.
Anyway, I think adding `useSoftFloat` matches the conversions better, I'll add it in another patch.

Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:1983]: https://github.com/llvm/llvm-project/commit/006f264b0343d0a27a95854e4f3aa610e5dac3bb

**Pull request merged**